templates/raspbian_cattlepi/resources/usr/share/initramfs-tools/scripts/cattlepi-base/helpers

```diff
@@ -414,3 +447,24 @@ cattlepi_bring_up_wlan() {
         return 1
 	fi
 }
+
+cattlepi_sdmount() {
+    # Explicitly mount the sdcard partition to /boot
+    # We don't give the caller any other option as of right
+    # now since this partition doesn't need to be mounted
+    # anywhere else and doing so could be dangerous
+    echo "Mounting sdcard to /boot"
+    mkdir /boot
+    umount /boot
```

Comment:
insert a **sync** after this umount

Reply:
Wouldn't we want to sync the data to disk *before* the umount? Or am I thinking about that wrong?

Reply:
Huh. Now that you explicitely ask about it i'm not that sure. 
On a "normal" filesystem (ie not vfat) the umount should flush all the pending writes to disk. Normally if you copy a large file to the card you would expect that upon unmount the kernel would wait for the writes to complete. What I have observed is that it does not and you end up with a corrupted file.

Now, Should the sync be before or after the umount. Techincally I think the right answer is before the unmount (but after any copy/write operation). You force everything to be written and when the sync completes the umount is a formality.
Why does it work this way? (after umount). I think when I was researching this I read somewhere that the umount for vfat is done in a lazy manner. What this means is that the kernel does the bookkeeping and after that the writes will be flushed out (eventually?? saw this by having it sleep for tens of seconds). So the sync after the umount actually forces it to do the writes now.
So lazy umount + sync == proper umount. 

I have a hard time finding the reference where the lazy behavior for vfat was documented now. I recall that after I did this it seemed to work each time so I never though about it too much (which is why I think it's pretty cool when someone else can verify/question your though process :+1: )

So to answer your question: it probably works both ways and it should probably be _before_ the umount. 
No, you are not thinkign about it wrong. Any sane filesystem would not need the sync to umount properly.

Reply:
Having all sorts of issues trying to get the bcrmfmac driver to load properly and the wlan0 interface to show up.  Still investigating and will continue to work on this.